xcdatamodeld not compiled for static framework

* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Created a pod with CoreData model included. In podspec specified `static_framework = true`.
Run `pod lib lint --allow-warnings` command

## What did you expect to happen?

Pod should pass validation

## What happened instead?

CoreData model wasn't compiled and classes for models weren't generated. 
Thus pod validation failed.
If I comment out `static_framework = true` then validation passes. But for some reasons I have to keep static_framework set to true (my actual pod depends on another static pod, so this makes me to keep my pod static too).
`pod lib lint` output:
```
 -> TestStorage (0.1.0)
    - WARN  | [iOS] swift: The validator used Swift `4.0` by default because no Swift version was specified. To specify a Swift version during validation, add the `swift_versions` attribute in your podspec. Note that usage of a `.swift-version` file is now deprecated.
    - ERROR | [iOS] xcodebuild: Returned an unsuccessful exit code.
    - NOTE  | xcodebuild:  note: Using new build system
    - NOTE  | [iOS] xcodebuild:  note: Planning build
    - NOTE  | [iOS] xcodebuild:  note: Constructing build description
    - NOTE  | [iOS] xcodebuild:  warning: Skipping code signing because the target does not have an Info.plist file. (in target 'App')
    - ERROR | [iOS] xcodebuild:  /Users/y.bogdanov/Dev/TestStorage/TestStorage/Classes/ReplaceMe.swift:3:19: error: use of undeclared type 'Employee'
    - ERROR | [iOS] xcodebuild:  /Users/y.bogdanov/Dev/TestStorage/TestStorage/Classes/ReplaceMe.swift:4:17: error: use of undeclared type 'Office'
    - ERROR | [iOS] xcodebuild:  /Users/y.bogdanov/Dev/TestStorage/TestStorage/Classes/ReplaceMe.swift:6:20: error: use of undeclared type 'Employee'
    - ERROR | [iOS] xcodebuild:  /Users/y.bogdanov/Dev/TestStorage/TestStorage/Classes/ReplaceMe.swift:6:41: error: use of undeclared type 'Office'

[!] TestStorage did not pass validation, due to 5 errors.
You can use the `--no-clean` option to inspect any issue.
```


## CocoaPods Environment

This happens with both CocoaPods 1.6.1 and 1.7.0.beta.3

### Stack

```
   CocoaPods : 1.7.0.beta.3
        Ruby : ruby 2.6.1p33 (2019-01-30 revision 66950) [x86_64-darwin18]
    RubyGems : 3.0.1
        Host : Mac OS X 10.14.4 (18E226)
       Xcode : 10.1 (10B61)
         Git : git version 2.17.2 (Apple Git-113)
Ruby lib dir : /Users/y.bogdanov/.rbenv/versions/2.6.1/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ cd9f8c8ff1138950954cf3b857b3532090642f72
```

### Installation Source

```
Executable Path: /Users/y.bogdanov/.rbenv/versions/2.6.1/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.4
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
```

## Project that demonstrates the issue
https://github.com/YuryBogdanov/TestStorage
I left `static_framework = true` line commented. Feel free to uncomment it to reproduce validation fail.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -16,6 +16,9 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   [Eric Amorde](https://github.com/amorde)
   [#8695](https://github.com/CocoaPods/CocoaPods/issues/8695)
 
+* Fix regression not compiling xcdatamodeld files in static frameworks  
+  [Paul Beusterien](https://github.com/paulb777)
+  [#8702](https://github.com/CocoaPods/CocoaPods/issues/8702)
 
 ## 1.7.0.beta.3 (2019-03-28)
 

--- a/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb
+++ b/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb
@@ -303,7 +303,7 @@ module Pod
               other_file_refs = project_file_references_array(other_source_files, 'other source')
               native_target.add_file_references(other_file_refs, nil)
 
-              next unless target.build_as_dynamic_framework?
+              next unless target.build_as_framework?
 
               filter_resource_file_references(file_accessor.resources.flatten) do |compile_phase_refs, resource_phase_refs|
                 native_target.add_file_references(compile_phase_refs, nil)

--- a/spec/unit/installer/xcode/pods_project_generator/pod_target_installer_spec.rb
+++ b/spec/unit/installer/xcode/pods_project_generator/pod_target_installer_spec.rb
@@ -731,6 +731,18 @@ module Pod
               resource.should.be.not.nil
             end
 
+            it 'adds framework resources to the static framework target' do
+              @pod_target.stubs(:build_type => Target::BuildType.static_framework)
+              @installer.install!
+              resources = @project.targets.first.resources_build_phase.files
+              resources.count.should > 0
+              resource = resources.find { |res| res.file_ref.path.include?('logo-sidebar.png') }
+              resource.should.be.not.nil
+
+              resource = resources.find { |res| res.file_ref.path.include?('en.lproj') }
+              resource.should.be.not.nil
+            end
+
             #--------------------------------------#
 
             describe 'with a scoped pod target' do

